packages/repository/src/decorators/model.ts

```diff
@@ -73,3 +75,26 @@ export function property(definition?: Partial<PropertyDefinition>) {
     Object.assign({}, definition),
   );
 }
+
+export namespace property {
+  export const array = function(
+    itemType: Function,
+    definition?: Partial<PropertyDefinition>,
+  ) {
+    return function(target: Object, propertyName: string) {
+      const propType = MetadataInspector.getDesignTypeForProperty(
+        target,
+        propertyName,
+      );
+      if (!Array.isArray(propType())) {
```

Comment:
The test should be `propType !== Array`.

Reply:
It wasn't letting me make that comparison before, though, that may have been due to another incorrect type. Let me try again.